tests/test_file_operations.py

```diff
@@ -0,0 +1,606 @@
+    os.remove(dest_file)
+    os.rmdir(dest_dir)
+
+
+def test_move_not_absolute(test_dir):
+    """Test that a ValueError is raised when a relative
+    path is given to the move operation"""
+
+    source_dir = os.path.join(test_dir, "from_here")
+    os.mkdir(source_dir)
+    dest_dir = os.path.join(test_dir, "to_here")
+    os.mkdir(dest_dir)
+
+    dest_file = ".."
+
+    source_file = pathlib.Path(source_dir) / "app_move.txt"
+    with open(source_file, "w+", encoding="utf-8") as dummy_file:
+        dummy_file.write("dummy")
+
+    parser = get_parser()
+    cmd = f"move {source_file} {dest_file}"
+    args = cmd.split()
+    ns = parser.parse_args(args)
+
+    with pytest.raises(ValueError) as ex:
+        file_operations.move(ns)
+    assert f"path {dest_file} must be absolute" in ex.value.args
+
+
+def test_remove_op_file(test_dir):
+    """Test the operation to delete a file"""
+
+    # Make a test file with dummy text
+    to_del = pathlib.Path(test_dir) / "app_del.txt"
+    with open(to_del, "w+", encoding="utf-8") as dummy_file:
+        dummy_file.write("dummy")
+
+    assert osp.exists(to_del)
+    with open(to_del, "r", encoding="utf-8") as dummy_file:
+        assert dummy_file.read() == "dummy"
+
+    parser = get_parser()
+    cmd = f"remove {to_del}"
+    args = cmd.split()
+    ns = parser.parse_args(args)
+
+    file_operations.remove(ns)
+
+    # Assert file has been deleted
+    assert not osp.exists(to_del)
+
+
+def test_remove_op_dir(test_dir):
+    """Test the operation to delete a directory"""
+
+    # Make a test file with dummy text
+    to_del = pathlib.Path(test_dir) / "dir_del"
+    os.mkdir(to_del)
+
+    parser = get_parser()
+    cmd = f"remove {to_del}"
+    args = cmd.split()
+    ns = parser.parse_args(args)
+
+    file_operations.remove(ns)
+
+    # Assert directory has been deleted
+    assert not osp.exists(to_del)
+
+
+def test_remove_op_bad_path(test_dir):
+    """Test that FileNotFoundError is raised when a bad path is given to the
+    operation to delete a file"""
+
+    to_del = pathlib.Path(test_dir) / "not_real.txt"
+
+    parser = get_parser()
+    cmd = f"remove {to_del}"
+    args = cmd.split()
+    ns = parser.parse_args(args)
+
+    with pytest.raises(FileNotFoundError) as ex:
+        file_operations.remove(ns)
+    assert "No such file or directory" in ex.value.args
+
+
+def test_remove_op_no_absolute():
+    """Test that ValueError is raised when a relative path
+    is given to the operation to delete a file"""
+
+    to_del = ".."
+
+    parser = get_parser()
+    cmd = f"remove {to_del}"
+    args = cmd.split()
+    ns = parser.parse_args(args)
+
+    with pytest.raises(ValueError) as ex:
+        file_operations.remove(ns)
+    assert f"path {to_del} must be absolute" in ex.value.args
+
+
+@pytest.mark.parametrize(
+    ["param_dict", "error_type"],
+    [
+        pytest.param(
+            {
+                "5": 10,
+                "FIRST": "SECOND",
+                "17": 20,
```

Comment:
this is carry over from previous tests. I can remove it if you think its irrelevant

Reply:
I'd say either assume a `dict[str, str]` or introduce a [type alias](https://github.com/CrayLabs/SmartSim/pull/615#discussion_r1681912000) so that we know exactly what you are expecting before we serialize and send a command over the wire